google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1134,6 +1134,11 @@ public SpannerOptions build() {
         this.setChannelConfigurator(ManagedChannelBuilder::usePlaintext);
         // As we are using plain text, we should never send any credentials.
         this.setCredentials(NoCredentials.getInstance());
+        // Default project id resolution might fail, but the emulator accepts any
+        // project id, so we can safely use a dummy id.
+        if (this.getProjectId() == null) {
```

Comment:
oops, may need more work as there's no such getter in ServiceOptions builder and the field is private https://github.com/googleapis/java-core/blob/375983090b3700b3fb6a1953626db7efca49cc51/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L127